### PR TITLE
RSDK-2828 - add documentation to motor.dart

### PR DESCRIPTION
**Pull request by clintpurser**
Adds documentation comments pulled and slightly modified from thepython sdk [motor.py](https://github.com/viamrobotics/viam-python-sdk/blob/main/src/viam/components/motor/motor.py)

### CODE REVIEW

File: lib/src/components/motor/motor.dart [hunk 1]
```diff
@@ -33,22 +33,71 @@ abstract class Motor extends Resource {
 
   Motor(this.name);
 
-  Future<void> setPower(double powerPct, {Map<String, dynamic>? extra});
```
Comment (clintpurser) [inline lib/src/components/motor/motor.dart:36]: driveby: I changed this to from `powerPct` to `power` to match the python sdk / documentation. seemed minor enough to include this rename here.

This is the only code change in the PR.

Comment (cheukt) [reply lib/src/components/motor/motor.dart:36]: would keep at powerPct

Comment (clintpurser) [reply lib/src/components/motor/motor.dart:36]: ✅ back to `powerPct`

File: lib/src/components/motor/motor.dart [hunk 2]
```diff
@@ -33,22 +37,71 @@ abstract class Motor extends Resource {
...
   Future<void> goFor(double rpm, double revolutions, {Map<String, dynamic>? extra});
 
+  /// Spin the [Motor] to the specified position (provided in revolutions from home/zero),
+  /// at the specified speed, in revolutions per minute.
+  /// Regardless of the directionality of the [rpm] this function will move
+  /// the [Motor] towards the specified position.
+  ///
+  /// Args:
+  ///     [rpm] (double): Speed at which the motor should rotate (absolute value).
+  ///     [positionRevolutions] (double): Target position relative to home/zero, in revolutions.
```
Comment (njooma) [inline lib/src/components/motor/motor.dart:66]: It's actually not super important to have these enumerated like this, because the flutter documentation generator is much better than the python one.

So while in python it's necessary to list out all args and return types, it actually looks bad in Flutter to do so:
> image

You can see that it doesn't format the args nicely and it just looks weird. 

Unlike in python where it formats like this:
> image

So I would remove all the `Args` and `Returns` lines.

**Review by njooma — CHANGES_REQUESTED** [review]

File: lib/src/components/motor/motor.dart (hunk 2 above)
Comment (clintpurser) [reply lib/src/components/motor/motor.dart:66]: Thanks for calling this out. I will remove those.

Comment (clintpurser) [reply lib/src/components/motor/motor.dart]: ✅ removed the Args and Returns from the documentation.

**Review by njooma — APPROVED** [review]

**Review by stuqdog — APPROVED** [review]

**Review by cheukt — APPROVED** [review]
